recipe/build.sh

```diff
@@ -1,12 +1,21 @@
 #!/bin/bash
 
 export HDF5_VERSION=${hdf5}
-export HDF5_DIR=${PREFIX}
+if [[ "$CONDA_BUILD_CROSS_COMPILATION" == "1" ]]; then
+  # load HDF5 library from _build_env
+  export HDF5_DIR=${BUILD_PREFIX}
+else
+  export HDF5_DIR=${PREFIX}
+fi
 export OPAL_PREFIX=${PREFIX}
 if [[ "$mpi" != "nompi" ]]; then
   export HDF5_MPI="ON"
 fi
 
+if [[ ${target_platform} == "osx-arm64" ]]; then
+  export CIBW_ARCHS_MACOS=arm64
```

Comment:
I think this line deserves a comment explaining why we are using a flag supposedly targeting a different build system

Reply:
I even forgot why I added it....

Reply:
Happy to add anything you suggest (presuming you can't directly commit to this branch), how about:
```suggestion
  # disable complex256 on macOS ARM64, see https://github.com/h5py/h5py/pull/2065
  export CIBW_ARCHS_MACOS=arm64
```

Reply:
great!